tensorflow/compiler/mlir/lite/utils/validators.cc

```diff
@@ -78,5 +78,19 @@ bool IsBroadcastableElementsAttrs(mlir::Attribute a, mlir::Attribute b) {
   return OpTrait::util::getBroadcastedType(a.getType(), b.getType()) != Type();
 }
 
+bool IsDimensionsDegenerateExceptLastOne(ArrayRef<int64_t> elements_shape) {
+  for (auto i = 0; i < elements_shape.size() - 1; i++) {
+    if (elements_shape[i] != 1) return false;
+  }
+  return true;
+}
+
+bool IsDimensionsDegenerateExceptLastOne(Attribute val) {
+  if (auto elements = val.dyn_cast<DenseElementsAttr>()) {
+    return IsDimensionsDegenerateExceptLastOne(elements.getType().getShape());
```

Comment:
This line should be rewritten in order to access the shape information if available.

For example,

if (auto ranked_type = elements.getType().cast<RankedTensorType>()) {
  return IsDimensionsDegenerateExceptLastOne(ranked_type.getShape());
}

Otherwise, it will cause crashing when the type is UnrankedTensorType.

Reply:
Good catch, I fixed it in 4ebd2633e870dd474fbc9b348d311a4ccdf20158